mm2src/coins/eth.rs

```diff
@@ -826,6 +838,28 @@ impl EthCoinImpl {
         sha256(&input).to_vec()
     }
 
+    /// The id used to differentiate payments on Etomic swap smart contract.
+    /// For Taker Etomic Swap contract taker_secret_hash should be provided.
+    pub(crate) fn etomic_swap_v2_id(
+        &self,
+        funding_time_lock: u32,
+        payment_time_lock: u32,
+        secret_hash: &[u8],
+    ) -> Vec<u8> {
+        let funding_timelock_bytes = funding_time_lock.to_le_bytes();
+        let payment_timelock_bytes = payment_time_lock.to_le_bytes();
+
+        let mut input =
+            Vec::with_capacity(funding_timelock_bytes.len() + payment_timelock_bytes.len() + secret_hash.len());
+        input.extend(
+            funding_timelock_bytes
+                .iter()
+                .chain(payment_timelock_bytes.iter())
+                .chain(secret_hash.iter()),
+        );
+        sha256(&input).to_vec()
+    }
```

Comment:
Got it, thanks for the clarification 
used old function https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/071a1f6ce428db2d995aa3b4d1bfc393bd2fd5eb

I thought that since the Taker sends the payment first in protocol V2, we could use the taker's secret hash for the etomic swap ID in V2. However, now I see that we follow general locking mechanism rules and since the taker's secret hash is used for refunds, we need to use the maker's secret hash for the swap id.

Reply:
> This approach adds some extra layer of entropy to the id creation, enhancing its uniqueness.

It did enhance uniqueness when we were using H160, but with sha256, I don't think it does. If we used the maker secret hash as the id, then we can remove the id from all the etomic swap contract functions such as the below one. But it might not be worth it as the saved gas might be negligible. Anyways, I am fine with using the old function.
```js
    function ethTakerPayment(
        bytes32 id,
        uint256 dexFee,
        address receiver,
        bytes32 takerSecretHash,
        bytes32 makerSecretHash,
        uint32 preApproveLockTime,
        uint32 paymentLockTime
    ) external payable {
```